gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineFormat.java

```diff
@@ -63,13 +63,13 @@ public void apply(Project project) {
 
             // necessary because SpotlessPlugin creates tasks in an afterEvaluate block
             Task formatTask = project.task("format");
-            if (eclipseFormattingEnabled(project) && !Files.exists(eclipseXml)) {
-                formatTask.dependsOn(project.getTasks().findByPath(":baselineUpdateConfig"));
-            }
             project.afterEvaluate(p -> {
-                Task spotlessApply = project.getTasks().getByName("spotlessApply");
-                formatTask.dependsOn(spotlessApply);
-                project.getTasks().withType(JavaCompile.class).configureEach(spotlessApply::mustRunAfter);
+                Task spotlessJava = project.getTasks().getByName("spotlessJava");
```

Comment:
There was also another slight bug, where we referenced the wrong spotless task. Turns out that the task doing all of the work is `spotlessJava` and that it uses the presence of other tasks in the task graph to determine how to behave https://github.com/diffplug/spotless/blob/master/plugin-gradle/src/main/java/com/diffplug/gradle/spotless/SpotlessPlugin.java#L99